platforms/documentation/docs/src/docs/userguide/dep-man/understanding-dependency-resolution/dependency_caching.adoc

```diff
@@ -0,0 +1,265 @@
+
+====
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-dynamic/kotlin",files="build.gradle.kts[tags=dynamic-version-cache-control]"]
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-dynamic/groovy",files="build.gradle[tags=dynamic-version-cache-control]"]
+====
+
+To change how long Gradle will cache the metadata and artifacts for a changing module, use:
+
+====
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-changing/kotlin",files="build.gradle.kts[tags=changing-module-cache-control]"]
+include::sample[dir="snippets/dependencyManagement/troubleshooting-cache-changing/groovy",files="build.gradle[tags=changing-module-cache-control]"]
+====
+
+== 3. Caching changing dependencies
+
+Gradle treats dependencies marked as "changing" (such as SNAPSHOT dependencies) differently from regular dependencies, refreshing them more frequently to ensure that you are always using the latest version.
+
+To declare a dependency as changing, you can set the `changing = true` attribute in your dependency declaration.
+This is useful for dependencies expected to change frequently without a new version number:
+
+[source,groovy]
+----
+dependencies {
+    implementation("com.example:my-library:1.0-SNAPSHOT") {
+        changing = true
+    }
+}
+----
+
+[[sec:offline-mode]]
+== 4. Using offline mode
+
+The `--offline` command-line switch instructs Gradle to use dependency modules from the cache, regardless of whether they are due to be checked again.
+When running with `offline`, Gradle will not attempt to access the network for dependency resolution.
+If the required modules are not in the dependency cache, the build will fail.
+
+[[sec:refreshing-dependencies]]
+== 5. Force-refreshing dependencies
+
+You can control the behavior of dependency caching for a distinct build invocation from the command line.
+Command line options help make a selective, ad-hoc choice for a single build execution.
+
+At times, the Gradle Dependency Cache can become out of sync with the actual state of the configured repositories.
+Perhaps a repository was initially misconfigured, or maybe a "non-changing" module was published incorrectly.
+To refresh all dependencies in the dependency cache, use the `--refresh-dependencies` option on the command line.
+
+The `--refresh-dependencies` option tells Gradle to ignore all cached entries for resolved modules and artifacts.
+A fresh resolve will be performed against all configured repositories, with dynamic versions recalculated, modules refreshed, and artifacts downloaded.
+However, where possible Gradle will check if the previously downloaded artifacts are valid before downloading again.
+This is done by comparing published SHA1 values in the repository with the SHA1 values for existing downloaded artifacts.
+
+- new versions of dynamic dependencies
+- new versions of changing modules (modules that use the same version string but can have different contents)
```

Comment:
This list doesn't have any preceding text explaining what it is.

Reply:
Removed.